Packs/PhishTank/Integrations/PhishTankV2/PhishTankV2.py

```diff
@@ -0,0 +1,307 @@
+import urllib3
+
+from CommonServerPython import *
+
+RESPONSE_LINE_LENGTH = 8
+
+urllib3.disable_warnings()
+
+''' CONSTANTS '''
+
+BASE_URL = 'http://data.phishtank.com'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+RELOAD_DATA_URL_SUFFIX = "data/online-valid.csv"
+
+''' CLIENT CLASS '''
+
+
+def handle_error(res: requests.models.Response):
+    if res.status_code == 404 or res.status_code == 509:
+        err_msg = f'PhishTankV2 - Error in API call {res.status_code} - {res.reason}'
+        return_error(err_msg)
+
+
+class Client(BaseClient):
+    """
+    Client to use in the PhisTankV2 integration. Overrides BaseClient.
+
+        Args:
+           proxy (bool): False if feed HTTPS server certificate will not use proxies, True otherwise.
+           insecure (bool): False if feed HTTPS server certificate should be verified, True otherwise.
+           verify (bool) : not insecure
+           fetch_interval_hours (str) : Database refresh interval (hours)
+   """
+
+    def __init__(self, proxy: bool, verify: bool, fetch_interval_hours: str):
+        super().__init__(proxy=proxy, verify=verify, base_url=BASE_URL)
+        self.fetch_interval_hours = fetch_interval_hours
+
+    def get_http_request(self, url_suffix: str):
+        result = self._http_request(
+            method='GET',
+            url_suffix=url_suffix,
+            resp_type="text",
+            error_handler=handle_error
+        )
+        return result
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+    """
+    data = reload(client)
+    response_was_empty = len(data.keys()) == 0
+    if response_was_empty:
+        return_error("Error - could not fetch PhishTankV2 database, "
+                     "API returned an empty response")
+    return 'ok'
+
+
+def context_not_loaded_yet(context: dict):
+    is_context_not_loaded = not context
```

Comment:
this name here as well - context is just a tool we're using, it isn't the essence.
What we're actually doing here is check if we ever reload the phishtank data.

Reply:
Done